sync_events.py

```diff
@@ -0,0 +1,222 @@
+
+        event = cls(
+            filename=slugify(cls._parse_title(api_event), max_length=50),
+            pretix_slug=pretix_slug,
+            is_deleted=False,
+            title=cls._parse_title(api_event),
+            event_date=cls._parse_date(api_event),
+            event_time=cls._parse_time(api_event),
+            event_registration="https://pretix.eu/correlaid/" + pretix_slug,
+            correlaidx=False,
+            languages=[],
+            tags=[],
+            description=cls._parse_description(api_event),
+        )
+
+        event.save()
+        return event
+
+    def update(self, api_event):
+        self.is_deleted = False
+        self.title = self._parse_title(api_event)
+        self.event_date = self._parse_date(api_event)
+        self.event_time = self._parse_time(api_event)
+        self.description = self._parse_description(api_event)
+        self.save()
+
+    def delete(self):
+        self.is_deleted = True
+        self.save()
+
+    @classmethod
+    def load(cls, filepath):
+        with open(filepath) as f:
+            contents = f.read()
+            _, contents = contents.split("---", 1)
+            front_matter, description = contents.split("---", 1)
+
+        front_matter = yaml.load(front_matter, Loader=yaml.FullLoader)
+        description = description.strip()
+
+        if front_matter.get("pretixSlug") is None:
+            # not a pretix event
+            return None
+
+        filename = filepath.name[: -len(filepath.suffix)]
+
+        return cls(
+            filename=filename,
+            pretix_slug=front_matter["pretixSlug"],
+            is_deleted=front_matter["isDeleted"],
+            title=front_matter["title"],
+            event_date=front_matter["eventDate"],
+            event_time=front_matter["eventTime"],
+            event_registration=front_matter["eventRegistration"],
+            correlaidx=front_matter["correlaidx"],
+            languages=front_matter["languages"],
+            tags=front_matter["tags"],
+            description=description,
+        )
+
+    def save(self):
+        front_matter = yaml.dump(
+            {
+                "pretixSlug": self.pretix_slug,
+                "isDeleted": self.is_deleted,
+                "title": self.title,
+                "eventDate": self.event_date,
+                "eventTime": self.event_time,
+                "eventRegistration": self.event_registration,
+                "correlaidx": self.correlaidx,
+                "languages": self.languages,
+                "tags": self.tags,
+            },
+            sort_keys=False,
+        )
+
+        content = f"---\n{front_matter}---\n\n{self.description}"
+
+        filepath = Path(f"content/en/events/{self.event_date[:7]}/{self.filename}.md")
+        filepath.parent.mkdir(parents=True, exist_ok=True)
+        with open(filepath, "w") as f:
+            f.write(content)
+
+    @staticmethod
+    def _parse_title(api_event):
+        # TODO langs
```

Comment:
TODO - probably need to discuss this one

Reply:
https://github.com/CorrelAid/hugo-website/pull/200#discussion_r701980692

i'll have a go at that soon

Reply:
I'll close this thread and just use https://github.com/CorrelAid/hugo-website/pull/200#discussion_r701980692